api/layer/multipart_upload.go

```diff
@@ -595,14 +595,67 @@ func (n *layer) reUploadFollowingParts(ctx context.Context, uploadParams UploadP
 	for _, part := range parts {
 		uploadParams.PartNumber = part.Number
 
-		if err = n.reUploadPart(ctx, uploadParams, part.OID, bktInfo, multipartInfo); err != nil {
-			return fmt.Errorf("reupload number=%d: %w", part.Number, err)
+		if len(part.Elements) > 0 {
+			if err = n.reUploadSegmentedPart(ctx, uploadParams, part, bktInfo, multipartInfo); err != nil {
+				return fmt.Errorf("reupload number=%d: %w", part.Number, err)
+			}
+		} else {
+			if err = n.reUploadPart(ctx, uploadParams, part.OID, bktInfo, multipartInfo); err != nil {
+				return fmt.Errorf("reupload number=%d: %w", part.Number, err)
+			}
 		}
 	}
 
 	return nil
 }
 
+func (n *layer) reUploadSegmentedPart(ctx context.Context, uploadParams UploadPartParams, part *data.PartInfo, bktInfo *data.BucketInfo, multipartInfo *data.MultipartInfo) error {
+	completePartPayload := make([]byte, 0, n.neoFS.MaxObjectSize())
+
+	for _, element := range part.Elements {
+		elementObj, err := n.objectGet(ctx, bktInfo, element.OID)
+		if err != nil {
+			return fmt.Errorf("get part oid=%s, emelent oid=%s: %w", part.OID.String(), element.OID.String(), err)
+		}
+
+		completePartPayload = append(completePartPayload, elementObj.Payload()...)
```

Comment:
10G object?

Reply:
This is a huge problem because we have to know the whole part size before uploading. I am thinking about trying to add some extra fields in the meta, to get this info it

Reply:
UPD: Replaced with io.Pipe. It should be less memory aggressive